source/vulkan/vulkan_impl_command_list.cpp

```diff
@@ -864,43 +865,169 @@ void reshade::vulkan::command_list_impl::resolve_texture_region(api::resource sr
+			uint32_t(blit_info.srcOffsets[1].y - blit_info.srcOffsets[0].y),
+			uint32_t(blit_info.srcOffsets[1].z - blit_info.srcOffsets[0].z) };
+
+		const VkExtent3D dst_copy_extent = {
+			uint32_t(blit_info.dstOffsets[1].x - blit_info.dstOffsets[0].x),
+			uint32_t(blit_info.dstOffsets[1].y - blit_info.dstOffsets[0].y),
+			uint32_t(blit_info.dstOffsets[1].z - blit_info.dstOffsets[0].z) };
+
+		VkImageResolve resolve_region;
+		resolve_region.srcSubresource = blit_info.srcSubresource;
+		resolve_region.srcOffset = blit_info.srcOffsets[0];
+		resolve_region.dstSubresource = blit_info.srcSubresource;
+		resolve_region.dstOffset = blit_info.srcOffsets[0];
+		resolve_region.extent = src_copy_extent;
+
+		VkImageSubresourceRange dst_subresource_range;
+		dst_subresource_range.aspectMask = resolve_region.dstSubresource.aspectMask;
+		dst_subresource_range.baseMipLevel = resolve_region.dstSubresource.mipLevel;
+		dst_subresource_range.levelCount = 1;
+		dst_subresource_range.baseArrayLayer = resolve_region.dstSubresource.baseArrayLayer;
+		dst_subresource_range.layerCount = resolve_region.dstSubresource.layerCount;
+
+		VkImageViewUsageCreateInfo usage_info = { VK_STRUCTURE_TYPE_IMAGE_VIEW_USAGE_CREATE_INFO };
+		usage_info.usage = VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT;
+
+		VkImageViewCreateInfo img_create_info = { VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO, &usage_info };
+		img_create_info.image = dst.handle;
+		img_create_info.viewType = VK_IMAGE_VIEW_TYPE_2D_ARRAY;
+		img_create_info.format = src_data->create_info.format;
+		img_create_info.subresourceRange = dst_subresource_range;
+
+		VkImageView dst_img_view;
+		if (vk.CreateImageView(_device_impl->_orig, &img_create_info, nullptr, &dst_img_view)) {
+			assert(false);
+			return;
+		}
 
-	region.dstOffset = { dst_x, dst_y, dst_z };
+		VkImageSubresourceRange src_subresource_range;
+		src_subresource_range.aspectMask = resolve_region.srcSubresource.aspectMask;
+		src_subresource_range.baseMipLevel = resolve_region.srcSubresource.mipLevel;
+		src_subresource_range.levelCount = 1;
+		src_subresource_range.baseArrayLayer = resolve_region.srcSubresource.baseArrayLayer;
+		src_subresource_range.layerCount = resolve_region.srcSubresource.layerCount;
 
-	vk.CmdResolveImage(_orig,
-		(VkImage)src.handle, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
-		(VkImage)dst.handle, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
-		1, &region);
+		img_create_info.image = src.handle;
+		img_create_info.subresourceRange = src_subresource_range;
+
+		VkImageView src_img_view;
+		if (vk.CreateImageView(_device_impl->_orig, &img_create_info, nullptr, &src_img_view)) {
+			assert(false);
+			return;
+		}
+
+		VkRenderingAttachmentInfo depth_attachment = { VK_STRUCTURE_TYPE_RENDERING_ATTACHMENT_INFO };
+		depth_attachment.imageView = src_img_view;
+		depth_attachment.imageLayout = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+		depth_attachment.resolveMode = VK_RESOLVE_MODE_AVERAGE_BIT;
```

Comment:
Implementations are only required to support `VK_RESOLVE_MODE_SAMPLE_ZERO_BIT`, `VK_RESOLVE_MODE_AVERAGE_BIT` is optional, so using it would require checking `VkPhysicalDeviceDepthStencilResolveProperties::supportedDepthResolveModes` (see https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceDepthStencilResolvePropertiesKHR.html). I think it's easier to just use sample 0 to avoid that.

Reply:
`VK_RESOLVE_MODE_AVERAGE_BIT` can be used if necessary, but would need to verify if the device actually supports it first then and fall back to `VK_RESOLVE_MODE_SAMPLE_ZERO_BIT` if not.